docs/utilities/idempotency.md

```diff
@@ -215,6 +217,70 @@ Resources:
     see 1WCU and 1RCU. Review the [DynamoDB pricing documentation](https://aws.amazon.com/dynamodb/pricing/) to
     estimate the cost.
 
+#### Using Redis as persistent storage layer
+
+##### Redis resources
+
+You need an existing Redis service before setting up Redis as persistent storage layer provider. You can also use Redis compatible services like [Amazon ElastiCache for Redis](https://aws.amazon.com/elasticache/redis/) as persistent storage layer provider.
+!!! tip "Tip:No existing Redis service?"
+If you don't have an existing Redis service, we recommend using DynamoDB as persistent storage layer provider.
+
+If you are using Redis you'll need to provide the Redis host, port, user and password as AWS Lambda environment variables.
+If you want to connect to a Redis cluster instead of a Standalone server, you need to enable Redis cluster mode by setting an AWS Lambda
+environment variable `REDIS_CLUSTER_MODE` to `true`
+In the following example, you can see a SAM template for deploying an AWS Lambda function by specifying the required environment variables.
+If you provide a [custom Redis client](#Customizing Redis client) you can omit the environment variables declaration.
+
+!!! warning "Warning: Avoid including a plain text secret directly in your template"
+This can infer security implications
+
+!!! warning "Warning: Large responses with Redis persistence layer"
+When using this utility with Redis your function's responses must be smaller than 512MB.
+Persisting larger items cannot might cause exceptions.
+
+```yaml hl_lines="9-13" title="AWS Serverless Application Model (SAM) example"
+Resources:
+  IdempotencyFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      CodeUri: Function
+      Handler: helloworld.App::handleRequest
+      Environment:
+        Variables:
+          REDIS_HOST: %redis-host%
+          REDIS_PORT: %redis-port%
+          REDIS_USER: %redis-user%
+          REDIS_SECRET: %redis-secret%
```

Comment:
See how Python managed the connection: https://github.com/aws-powertools/powertools-lambda-python/pull/2567/files#diff-9aa1f6d28c9806e244b0bf7a7263e02efb02cf4986da20d1b8fc6bb931097884R48

Reply:
With [0c1cc38](https://github.com/aws-powertools/powertools-lambda-java/pull/1513/commits/0c1cc38adaaebb10bf24a7dfdb76523cc9279775) the option to provide a [custom JedisClientConfig](https://github.com/aws-powertools/powertools-lambda-java/pull/1513/files#diff-13d12f68518bb8099731697346131611f57cc1e7c5a2b77276401dab47ce0fb0R86) is added. 
It can be used as described in the [documentation](https://github.com/aws-powertools/powertools-lambda-java/pull/1513/files#diff-e612f95a3dbd33cabab2e49899653b8cc5d5e809bfad45aa54cdee41d22e5808R972)